air/src/options.rs

```diff
@@ -383,7 +383,11 @@ impl PartitionOptions {
             self.min_partition_size as usize,
         );
 
-        base_elements_per_partition.div(E::EXTENSION_DEGREE)
+        base_elements_per_partition.div_ceil(E::EXTENSION_DEGREE)
```

Comment:
We may need to make this a bit more sophisticated because I think it will produce rather suboptimal results in some situations. For example:

- `num_columns = 7`
- `E::EXTENSION_DEGREE = 3`
- `self.num_partitions = 4`
- `self.min_partition_size = 8`

If I did my math correctly, with this approach, we'd get partition size of 3 columns which would imply 9 base field columns per partition. This would require 2 permutations of the hash function in each partition.

The previous approach would have actually resulted in a better outcome here (i.e., partition size 2, so so the 4 partitions would have 2, 2, 2, 1 columns). But this result would have been technically incorrect because we'd have 6 base field elements per partition and this would be smaller than `min_partition_size`.

Maybe instead of `min_partition_size` we should be specifying the number of base elements that can be absorbed per permutation and then we can adjust this algorithm to output more optimal results.